scimono-server/src/main/java/com/sap/scimono/helper/UnnecessarySchemasEraser.java

```diff
@@ -0,0 +1,25 @@
+
+package com.sap.scimono.helper;
+
+import com.sap.scimono.entity.Resource;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class UnnecessarySchemasEraser<T extends Resource<T>> {
+
+  public T eraseAllUnnecessarySchemas(final T resource, final String coreSchema) {
+
+    Set<String> schemasWithProvidedAttributes = resource.getExtensions().keySet();
+    Predicate<String> coreSchemaOrCustomSchemaWithProvidedAttributes = schema -> schema.equals(coreSchema)
+        || schemasWithProvidedAttributes.contains(schema);
```

Comment:
You nicely handle the case when schema like `urn:ietf:params:scim:schemas:core:2.0:Group` is part of the `schemas` attribute to erase it. What about the case when this schema is part of the body (part of `resource.getExtensions()`)?

if it is present in both places, such a request will pass and only the `schemas` occurrence will be erased.
Also, I think such a static transformations and validations  (without requirment of backend API) are ideal for handling when jackson mapping is processed. 
Look at the constructor of the `Resource` class and `@JsonAnySetter`  annotated method. This way you avoid additional copy of the user object.